### ISSUE TITLE
automatic registration in the vertx ServiceDiscovery

### ISSUE THREAD

**Issue by fiorenzino** [FroMage/redpipe#9]
Another suggestion:

the automatic registration of rest service on ServiceDiscovery should be interesting.

for example:
@ServiceDiscovery("api.name", "host", port, "root")
@Path("/")
public FlowerService {

}

translated in vertx:
 Record record = HttpEndpoint.createRecord(name, host, port, root, new JsonObject().put("api.name", name));
serviceDiscovery = ServiceDiscovery.create(vertx, new ServiceDiscoveryOptions().setBackendConfiguration(config()));
serviceDiscovery.publish(record, ar ->
         {
            if (ar.succeeded())
            {
               registeredRecords.add(record);
               logger.info("Service <" + ar.result().getName() + "> published");
               future.complete();
            }
            else
            {
               future.fail(ar.cause());
            }
         });

what do you think about it?

**Comment by FroMage** [comment]
That's very interesting, except I think the `host`/`port` parts should come from the current config, no? It seems error-prone to specify them in the code when they are deployment-dependent. I guess the same could be said of the `root`.

But certainly, it would be useful to declare services with an annotation. I think the name should be enough, the rest we can get from the config.

**Comment by fiorenzino** [comment]
Yes, the host/port/root parts are always derived from server config and @Path annotations. 
But in vertx you must declare that.

**Comment by FroMage** [comment]
Sure, but they don't belong in the annotation if the framework can detect them. Unless they're not required/defaulted.

**Comment by FroMage** [comment]
Yes this sounds great.
